Library/Homebrew/extend/ARGV.rb

```diff
@@ -125,7 +125,7 @@ def git?
   end
 
   def homebrew_developer?
-    include?("--homebrew-developer") || !ENV["HOMEBREW_DEVELOPER"].nil?
+    !ENV["HOMEBREW_DEVELOPER"].nil?
```

Comment:
I didn't even realize how misplaced this now looks. On the bright side, this keep the required changes to a minimum. I'm a bit torn on how to best resolve this, as moving this to `ENV` doesn't feel appropriate (the extensions there are mostly concerned with the build process).

Maybe something like `Config` could be created for things that can be passed both via arguments and the environment? The environment variables are effectively configuration, we just don't seem to like an approach where we would put them into a file, something a lot of software allows.